crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -925,31 +987,57 @@ impl CallChainLayout {
                 // ```
                 ExprRef::Call(ast::ExprCall { func: inner, .. })
                 | ExprRef::Subscript(ast::ExprSubscript { value: inner, .. }) => {
+                    // We preserve these parentheses so don't recurse
+                    // e.g. (a)[0].x().y().z()
+                    //         ^stop here
+                    if is_expression_parenthesized(inner.into(), comment_ranges, source) {
+                        break;
+                    }
+
+                    was_in_call_like = true;
```

Comment:
Why aren't we adding incrementing `call_like_count` in this branch? It might help to give `was_in_call_like` a more specific name or document its purpose when declaring the variable.